### PR TITLE
Adding memory confidentiality tests

### PR DESCRIPTION
**Pull request by spmcmillan**
JIRA: https://warthogs.atlassian.net/browse/PEK-1285

Adding tests to verify confidentiality with TDX and verify lack of confidentiality without TDX.

### CODE REVIEW

File: tests/tests/test_guest_memory.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import subprocess
+
+import Qemu
+from common import *
+
+MEMORY_FILE='memory.bin'
+CHECK_EXEC='export INCLEAR=6150079244'
+
+def check_for_string(filename, needle):
+    cs = subprocess.run(['sudo', 'grep', needle, filename], check=True, capture_output=True)
```
Comment (hector-cao) [inline tests/tests/test_guest_memory.py:30]: with `check=True`, the call will throw an exception if the string is not present in the file, that will make the test `test_guest_memory_confidentiality_tdx` fails and we do not want that, do we ?

**Review by hector-cao — CHANGES_REQUESTED** [review]

File: tests/tests/test_guest_memory.py (hunk 1 above)
Comment (spmcmillan) [reply tests/tests/test_guest_memory.py]: Yes, thanks for catching that.

Comment (spmcmillan) [reply tests/tests/test_guest_memory.py]: fixed in https://github.com/canonical/tdx/pull/235/commits/ea47d4ddcd72bba77caaf26e041bcc8cf3132d0e

**Review by hector-cao — COMMENTED** [review]
LGTM, do you plan to add these tests into checkbox in this PR or in an other upcoming PR ?

**Comment by spmcmillan** [comment]
> LGTM, do you plan to add these tests into checkbox in this PR or in an other upcoming PR ?

Just added it and tested.

**Review by hector-cao — APPROVED** [review]
LGTM, awesome work

**Pull request merged**